sphinx_gallery/block_parser.py

```diff
@@ -271,10 +287,11 @@ def finalize_block(mode, block):
                     else:
                         block.append(text)
                 else:
-                    block.append(self.continue_text.search(text).group(1))
+                    block.append(self.continue_text.search(text).group(1))  # type: ignore[union-attr]
```

Comment:
Mypy complains that search may return `Match | None`, and for the None case the `.group(1)` dereferencing would fail. We haven't observed that behavior at runtime. I haven't completely unraveled the logic here, so in theory there could be edge cases that crash. But let's not bother for now.